--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -59,27 +59,6 @@
 set(TR_SCM_REVISION_FILE "${CMAKE_SOURCE_DIR}/REVISION")
 set(TR_SCM_REVISION_RELIABLE ON)
 
-if(IS_DIRECTORY ${CMAKE_SOURCE_DIR}/.svn)
-    find_package(Subversion)
-    if(Subversion_FOUND)
-        Subversion_WC_INFO(${CMAKE_SOURCE_DIR} TR_SVN)
-        set(TR_SCM_REVISION "${TR_SVN_WC_REVISION}")
-    endif()
-elseif(IS_DIRECTORY ${CMAKE_SOURCE_DIR}/.git)
-    find_package(Git)
-    if(GIT_FOUND)
-        execute_process(
-            COMMAND
-                ${GIT_EXECUTABLE} rev-list --max-count=1 --abbrev-commit HEAD
-            WORKING_DIRECTORY
-                ${CMAKE_SOURCE_DIR}
-            OUTPUT_VARIABLE
-                TR_SCM_REVISION
-            OUTPUT_STRIP_TRAILING_WHITESPACE
-        )
-    endif()
-endif()
-
 if("${TR_SCM_REVISION}" STREQUAL "" AND EXISTS "${TR_SCM_REVISION_FILE}")
     file(READ "${TR_SCM_REVISION_FILE}" TR_SCM_REVISION)
     string(STRIP "${TR_SCM_REVISION}" TR_SCM_REVISION)
@@ -87,25 +66,7 @@
 
 if("${TR_SCM_REVISION}" STREQUAL "")
     # Give up and check the source files
-    set(TR_SCM_REVISION 0)
-    set(TR_SCM_REVISION_RELIABLE OFF)
-    file(GLOB_RECURSE TR_ALL_SOURCES RELATIVE ${CMAKE_SOURCE_DIR} *.cc *.[chm] *.po)
-    foreach(F ${TR_ALL_SOURCES})
-        file(STRINGS ${F} F_ID REGEX "\\$Id:")
-        if(F_ID MATCHES "\\$Id: [^ ]+ ([0-9]+) " AND CMAKE_MATCH_1 GREATER TR_SCM_REVISION)
-            set(TR_SCM_REVISION ${CMAKE_MATCH_1})
-        endif()
-    endforeach()
-endif()
-
-if("${TR_SCM_REVISION}" STREQUAL "")
-    set(TR_SCM_REVISION 0)
-else()
-    if(TR_SCM_REVISION_RELIABLE)
-        file(WRITE "${TR_SCM_REVISION_FILE}" "${TR_SCM_REVISION}")
-    else()
-        file(REMOVE "${TR_SCM_REVISION_FILE}")
-    endif()
+    message(FATAL_ERROR "Unable to get revision from ${TR_SCM_REVISION_FILE}.")
 endif()
 
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
